clients/src/main/java/org/apache/kafka/common/requests/DescribeConfigsRequest.java

```diff
@@ -17,70 +17,26 @@
 
 import java.nio.ByteBuffer;
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.Objects;
-
-import static org.apache.kafka.common.protocol.types.Type.BOOLEAN;
-import static org.apache.kafka.common.protocol.types.Type.INT8;
-import static org.apache.kafka.common.protocol.types.Type.STRING;
+import java.util.stream.Collectors;
 
 public class DescribeConfigsRequest extends AbstractRequest {
 
-    private static final String RESOURCES_KEY_NAME = "resources";
-    private static final String INCLUDE_SYNONYMS = "include_synonyms";
-    private static final String RESOURCE_TYPE_KEY_NAME = "resource_type";
-    private static final String RESOURCE_NAME_KEY_NAME = "resource_name";
-    private static final String CONFIG_NAMES_KEY_NAME = "config_names";
-
-    private static final Schema DESCRIBE_CONFIGS_REQUEST_RESOURCE_V0 = new Schema(
-            new Field(RESOURCE_TYPE_KEY_NAME, INT8),
-            new Field(RESOURCE_NAME_KEY_NAME, STRING),
-            new Field(CONFIG_NAMES_KEY_NAME, ArrayOf.nullable(STRING)));
-
-    private static final Schema DESCRIBE_CONFIGS_REQUEST_V0 = new Schema(
-            new Field(RESOURCES_KEY_NAME, new ArrayOf(DESCRIBE_CONFIGS_REQUEST_RESOURCE_V0), "An array of config resources to be returned."));
-
-    private static final Schema DESCRIBE_CONFIGS_REQUEST_V1 = new Schema(
-            new Field(RESOURCES_KEY_NAME, new ArrayOf(DESCRIBE_CONFIGS_REQUEST_RESOURCE_V0), "An array of config resources to be returned."),
-            new Field(INCLUDE_SYNONYMS, BOOLEAN));
-
-    /**
-     * The version number is bumped to indicate that on quota violation brokers send out responses before throttling.
-     */
-    private static final Schema DESCRIBE_CONFIGS_REQUEST_V2 = DESCRIBE_CONFIGS_REQUEST_V1;
-
-    public static Schema[] schemaVersions() {
-        return new Schema[]{DESCRIBE_CONFIGS_REQUEST_V0, DESCRIBE_CONFIGS_REQUEST_V1, DESCRIBE_CONFIGS_REQUEST_V2};
-    }
-
     public static class Builder extends AbstractRequest.Builder<DescribeConfigsRequest> {
-        private final Map<ConfigResource, Collection<String>> resourceToConfigNames;
-        private boolean includeSynonyms;
+        private final DescribeConfigsRequestData data;
 
-        public Builder(Map<ConfigResource, Collection<String>> resourceToConfigNames) {
+        public Builder(DescribeConfigsRequestData data) {
             super(ApiKeys.DESCRIBE_CONFIGS);
-            this.resourceToConfigNames = Objects.requireNonNull(resourceToConfigNames, "resourceToConfigNames");
-        }
-
-        public Builder includeSynonyms(boolean includeSynonyms) {
-            this.includeSynonyms = includeSynonyms;
-            return this;
-        }
-
-        public Builder(Collection<ConfigResource> resources) {
-            this(toResourceToConfigNames(resources));
+            this.data = data;
```

Comment:
As we remove the other constructors, `toResourceToConfigNames` is not called anymore. Let's delete it